src/methods.jl

```diff
@@ -77,3 +74,63 @@ julia> subtract_overscan(frame, (:, 4:5), dims = 2)
 * [`subtract_overscan!`](@ref)
 """
 subtract_overscan(frame::AbstractArray, idxs; dims = axes_min_length(idxs)) = subtract_overscan!(deepcopy(frame), idxs, dims = dims)
+
+
+"""
+    flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+
+In place version of [`flat_correct`](@ref)
+"""
+function flat_correct!(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+    size(frame) != size(flat_frame) && error("size of frame and flat frame are not same")
+    !(norm_value === nothing) && norm_value < 0 && error("norm_value must be greater than 0")
+
+    !(min_value === nothing) && @. flat_frame[flat_frame < min_value] = min_value
+
+    if !(norm_value === nothing)
+        flat_frame ./= norm_value
+    else
+        flat_frame ./= mean(flat_frame)
+    end
+
+    frame ./= flat_frame
+    return frame
+end
+
+
+"""
+    flat_correct(frame::AbstractArray, flat_frame::AbstractArray; min_value = nothing, norm_value = nothing)
+
+Correct `frame` for non-uniformity using the calibrated `flat_frame`.
+
+`min_value` is minimum value of `flat_frame`, an entry less than `min_value` is converted to `min_value` in `flat_frame`.
+`norm_value` is normalising constant for `flat_frame`, if set to `nothing` then `mean` of `flat_frame` is used for normalising.
+
+!!! note
+    The function [`flat_correct`](@ref) has a division step of `frame` by `flat_frame`, if `flat_frame` has a zero entry then
+    the returned value is either `Inf` (dividend `frame` element is non-zero) or `NaN` (dividend `frame` element is zero).
```

Comment:
A little awkward to reference the same docstring within the docstring-

"This function may introduce non-finite values if `flat_frame` contains values very close to `0` due to dividing by zero. The default behavior will return `Inf` if the `frame` value is non-zero, and `Nan` if the `frame` value is `0`."